File: 14-ExOh/14.hs
--Check if 'x' and 'o''s within a string are balanced in number
exOh :: String -> Bool
exOh str = (foldr (\x acc -> acc + if x == 'x' then 1 else -1) 0 str) == 0 

main = do
  print $ exOh "xxooxoox" == True
  print $ exOh "xxoox"    == False

File: 14-ExOh/14.js
//Check if 'x' and 'o''s within a string are balanced in number
function ExOh(str){ 
  return str.split('').reduce(function(acc,c){
    return acc + (c === 'x' ? 1 : -1)
  },0) === 0
}

console.log(ExOh("xxooxoox") === true)
console.log(ExOh("ooxox")    === false)